@here/harp-mapview-decoder/test/DataProviderTest.ts

```diff
@@ -60,4 +71,44 @@ describe("DataProvider", function() {
         dataProvider.unregister(clients[1]);
         expect(disposeSpy.calledOnce).is.true;
     });
+
+    it("calls connect only on first registered client", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+        const connectSpy = sinon.spy(dataProvider, "connect");
+
+        await dataProvider.register(clients[0]);
+        expect(connectSpy.calledOnce).is.true;
+
+        await dataProvider.register(clients[0]);
+        await dataProvider.register(clients[1]);
+        expect(connectSpy.calledOnce).is.true;
+    });
+
+    it("calls connect only on first registered client", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+        const connectSpy = sinon.spy(dataProvider, "connect");
+
+        dataProvider.register(clients[0]);
+        dataProvider.register(clients[1]);
+        expect(connectSpy.calledOnce).is.true;
+    });
+
+    it("subsequent register operations wait for connect", async function() {
```

Comment:
I am not sure, but would it work if while awaiting the Promise, the client gets unregistered and another or same one registered, or would this need disposal of the Promise: 

const promise0 = dataProvider.register(clients[0]);
dataProvider.unregister(clients[0]);
const promise0 = dataProvider.register(clients[1]);

Reply:
The current implementation does not reject the promise if the client is un-registered, for which I added tests. Currently, all clients get the same resolve (since they are using the same promise), even those that have been un-registered. 
There is also the issue of DataProviders that have their only client un-registered, then their dispose() method is called.
I am not sure if these cases need an improved specification.

Reply:
@FraukeF  that example you mentioned would dispose the dataProvider.

I guess also the promise should be rejected, @StefanDachwitz is that possible? Maybe the `connect()` method can return the Promise + the resolve / reject functions and you can reject the promise when the `dispose()` method is called?

Reply:
@nzjony The question is, if this is in the scope of the ticket.